prow/gerrit/adapter/adapter.go

```diff
@@ -73,25 +86,54 @@ var gerritMetrics = struct {
 		// Hopefully this isn't excessive enough to cause metric scraping issues.
 		"repo",
 	}),
+	triggerHelpLatency: prometheus.NewHistogramVec(prometheus.HistogramOpts{
+		Name:    "gerrit_trigger_help_latency",
+		Help:    "Histogram of seconds between triggering event (help) and ProwJob creation time.",
+		Buckets: []float64{5, 10, 20, 30, 60, 120, 180, 300, 600, 1200},
+	}, []string{
+		"org",
+	}),
+	processSingleChangeDuration: prometheus.NewHistogramVec(prometheus.HistogramOpts{
+		Name:    "gerrit_process_single_change_duration",
+		Help:    "Histogram of seconds spent processing a single gerrit change, by instance and repo.",
+		Buckets: []float64{5, 10, 20, 30, 60, 120, 180, 300, 600, 1200},
+	}, []string{
+		"org",
+		"repo",
+	}),
 	changeProcessDuration: prometheus.NewHistogramVec(prometheus.HistogramOpts{
 		Name:    "gerrit_instance_process_duration",
-		Help:    "Histogram of seconds spent processing a single gerrit instance.",
+		Help:    "Histogram of seconds spent processing changes, by instance and repo. This measures the portion of a sync after we've queried for changes.",
 		Buckets: []float64{5, 10, 20, 30, 60, 120, 180, 300, 600, 1200, 3600},
 	}, []string{
-		"org",
+		"org", "repo",
 	}),
 	changeSyncDuration: prometheus.NewHistogramVec(prometheus.HistogramOpts{
 		Name:    "gerrit_instance_change_sync_duration",
-		Help:    "Histogram of seconds spent syncing changes from a single gerrit instance.",
+		Help:    "Histogram of seconds spent syncing changes from a single gerrit instance or repo. Includes gerrit_repo_query_duration and gerrit_instance_process_duration.",
 		Buckets: []float64{5, 10, 20, 30, 60, 120, 180, 300, 600, 1200, 3600},
-	}, []string{"org"}),
+	}, []string{"org", "repo"}),
+	gerritRepoQueryDuration: prometheus.NewHistogramVec(prometheus.HistogramOpts{
+		Name:    "gerrit_repo_query_duration",
+		Help:    "Histogram of seconds spent querying a repo's changes. Includes time spent for rate limiting ourselves.",
+		Buckets: []float64{5, 10, 20, 30, 60, 120, 180, 300},
+	}, []string{"org", "repo", "result"}),
+	pickupChangeLatency: prometheus.NewHistogramVec(prometheus.HistogramOpts{
+		Name:    "gerrit_pickup_change_latency",
+		Help:    "Histogram of seconds a change had to wait after its creation but before it was picked up for processing by a worker thread.",
```

Comment:
I used the word "change" here to mean the query result, not the time that the CL was created (which is how I think you read my wording?). IOW I meant to say what you just said. Will reword because the word "change" is overloaded.